spec/models/manageiq/providers/microsoft/infra_manager/provision_workflow_spec.rb

```diff
@@ -16,6 +16,27 @@
     described_class.new({}, admin.userid)
   end
 
+  describe "#load_hosts_vlans" do
+    let(:hosts)       { [FactoryGirl.create(:host_with_ref)] }
+    let(:lan_parent)  { FactoryGirl.create(:lan) }
+    let(:lan)         { FactoryGirl.create(:lan) }
+    let(:switch)      { FactoryGirl.create(:switch) }
+
+    before do
+      stub_dialog
+      lan.update_attributes(:switch_id => switch.id, :parent_id => lan_parent.id)
```

Comment:
Can this just be part of the `let(:lan)`?

```ruby
let(:lan)         { FactoryGirl.create(:lan, :switch => switch, :parent => lan_parent) }
```

Reply:
@NickLaMuro - To both your comments, 'yes', but would need to be a separate PR off of manageiq.  The `lan` factory currently doesn't allow for traits, and there is no `host_switch` factory.  Unless we can add overridden factories to provider gems?